### ISSUE TITLE
'AgentPPO' object has no attribute 'traj_list'

### ISSUE THREAD

**Issue by HarryLukKF** [AI4Finance-Foundation/ElegantRL#181]
There is an error when I try to excute the train function in Demo_MultiCrypto_Trading.ipynb

AttributeError: 'AgentPPO' object has no attribute 'traj_list'
![image](https://user-images.githubusercontent.com/74358636/176986404-418420e3-162f-4609-a04f-dec5007d496e.png)

Please kindly advise how should I fix this error? Many thanks in advance

**Comment by AlexLiddiard** [comment]
Also encountered this issue when running Demo_MultiCrypto_Trading.ipynb
I spent quite a long time debugging it and finally got it working by changing these things:

1. elegantrl/agents/AgentBase.py line 43, insert:
```
self.traj_list = [
	[list() for _ in range(4 if self.if_off_policy else 5)]
	for _ in range(self.env_num)
]
```
2. elegantrl/agents/AgentPPO.py line 88, change to:
`return self.convert_trajectory(traj_list, last_done), step_i`

3. elegantrl/train/evaluator.py line 60, comment it out

4. finrl-meta/agents/elegantrl_models.py line 112, change to:
`environment.cash`

5. finrl-meta/agents/elegantrl_models.py line 114 change to:
`environment.price_array[environment.time] * environment.stocks`

**Comment by altosssss** [comment]
> Also encountered this issue when running Demo_MultiCrypto_Trading.ipynb I spent quite a long time debugging it and finally got it working by changing these things:
> 
> 1. elegantrl/agents/AgentBase.py line 43, insert:
> 
> ```
> self.traj_list = [
> 	[list() for _ in range(4 if self.if_off_policy else 5)]
> 	for _ in range(self.env_num)
> ]
> ```
> 
> 2. elegantrl/agents/AgentPPO.py line 88, change to:
>    `return self.convert_trajectory(traj_list, last_done), step_i`
> 3. elegantrl/train/evaluator.py line 60, comment it out
> 4. finrl-meta/agents/elegantrl_models.py line 112, change to:
>    `environment.cash`
> 5. finrl-meta/agents/elegantrl_models.py line 114 change to:
>    `environment.price_array[environment.time] * environment.stocks`

Thanks Alex. Wondering if you tried run it on Colab?

**Comment by bassemfg** [comment]
> > Also encountered this issue when running Demo_MultiCrypto_Trading.ipynb I spent quite a long time debugging it and finally got it working by changing these things:
> > 
> > 1. elegantrl/agents/AgentBase.py line 43, insert:
> > 
> > ```
> > self.traj_list = [
> > 	[list() for _ in range(4 if self.if_off_policy else 5)]
> > 	for _ in range(self.env_num)
> > ]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > 2. elegantrl/agents/AgentPPO.py line 88, change to:
> >    `return self.convert_trajectory(traj_list, last_done), step_i`
> > 3. elegantrl/train/evaluator.py line 60, comment it out
> > 4. finrl-meta/agents/elegantrl_models.py line 112, change to:
> >    `environment.cash`
> > 5. finrl-meta/agents/elegantrl_models.py line 114 change to:
> >    `environment.price_array[environment.time] * environment.stocks`
> 
> Thanks Alex. Wondering if you tried run it on Colab?

Yes, I am getting the same error both on ubuntu and google Colab

**Comment by bassemfg** [comment]
Also, if when I try other agents like ddpg or td3 or sac, I get another error in the same line saying that __init__() expects 4 parameters and got 5. I spent a couple of hours trying to debug it but no luck. Anyone has any ideas?

**Comment by shixun404** [comment]
I modified the question related to 'traj_list'. The file ' demo_A2C_PPO.py ' and ' demo_PPO_H.py ' could run correctly now!

**Comment by shixun404** [comment]
By the way, demo_DDPG_TD3_SAC.py and demo_DDPG_H.py run correctly as well~
